torch/_refs/__init__.py

```diff
@@ -4482,6 +4483,32 @@ def scalar_tensor(
     device = device if device is not None else torch.device("cpu")
     return prims.scalar_tensor(a, dtype=dtype, device=device)
 
+def check_for_1d_list_with_symints(a):
+    assert isinstance(a, Iterable)
+    len_ = None
+    for item in a:
+        if isinstance(item, Iterable):
+            return False
+        if isinstance(item, torch.SymInt):
+            return True
+    return False
+
+
+def torch_tensor_r(
+    a: Any,
+    *,
+    dtype: Optional[torch.dtype] = None,
+    device: Optional[torch.device] = None,
+    pin_memory: bool = False,
+    requires_grad: bool = False,
+) -> TensorLikeType:
+    utils.check_pin_memory(pin_memory)
+    # dtype = dtype if dtype is not None else utils.type_to_dtype(type(a))
+    device = device if device is not None else torch.device("cpu")
+    if isinstance(a, torch.SymInt) or check_for_1d_list_with_symints(a):
```

Comment:
You can support ND lists; just flatten the list before passing it to `int_tensor`

Reply:
I can but there are some checks we do to ensure data has a shape that's ok (consistent length of data with each dim). Skipped it for now but can add a similar function in python